### DISCUSSION TITLE
Bootstrap to Quasar question

### DISCUSSION THREAD

**Discussion by jonmor51** [quasarframework/quasar#11458] [General - Components / Directives / etc] [answered]
As a learning exercise, I am redoing with Quasar a little BootstrapVue site I made. Most of the time I've been able to find equivalents so far. But I can't see the Quasar way of doing an input with a prepended select which I have in one place. If you go to https://runcalc.jmsd.co.uk/ and select 'Pace Calculator' you will be able to see what I mean. The Distance input has a select in front of it which allows the user to select a standard running distance (which then populates the input). This isn't a standard select type, though, as it doesn't display the selected value and just shows the down arrow. Is there an equivalent way of doing something like this with Quasar?

**Comment by metalsadman** [comment, marked as answer]
use prepend or append slots with q-menu. https://codepen.io/metalsadman/pen/gOpMMxz

**Reply by jonmor51** [reply]
Thanks, that's very close to what I want. The only difference is that in the BootstrapVue version the prepended popup only shows a down arrow rather than the text of the selected item (which is important because a user may change the value via the input after selecting it from the popup). Simply setting `label=""` gets me part way there but doesn't look quite right somehow. I'll have a play, though, I'm sure it's just a case of some CSS tweaks.

LATER: Removing the label attribute altogether and setting a left padding of 5px for the button gives me something which works very well.

**Reply by metalsadman** [reply]
it's just a sample showing you that it can be done, it's up to you how to make it better and suit your needs. refer to the docs for more info about the components used in the codepen. good luck.

**Comment by pdanpdan** [comment]
You can take a look here: #7516
